app/Http/Middleware/Meetinga.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Database\Eloquent\Model;
+use EscolaLms\Auth\Services\Contracts\UserGroupServiceContract;
+use EscolaLms\Auth\Services\Contracts\UserServiceContract;
+use EscolaLms\Auth\Dtos\UserSaveDto;
+use EscolaLms\Auth\Dtos\UserUpdateSettingsDto;
+use EscolaLms\Auth\Models\Group;
+use Ramsey\Collection\Map\TypedMap;
+use Illuminate\Support\Facades\App;
+
+use Closure;
+
+class Meetinga
+{
+
+    private function createOrFindGroups(Model $mainGroup, array $groups): array
+    {
+
+        return array_map(
+            function ($group) use ($mainGroup) {
+                return Group::firstOrCreate([
+                    'name' => $group['name'],
+                    'registerable' => true,
+                    'parent_id' => $mainGroup->id
+                ]);
+            },
+            $groups
+        );
+    }
+
+    private function getUserCertificate($data): array
+    {
+
+        $slug = $data['registered_events'][0]['slug'];
+
+        $postResponse = Http::withHeaders([
+            'App-Version' => config('meetinga.app-version'),
+            'Api-Key' => config('meetinga.api-key'),
+            'Authorization' => "JWT " . $data['token']
+        ])->get(config('meetinga.url') . '/' . $slug . '/pl/api/awf/certificate/');
+
+        if ($postResponse->ok()) {
+            return $postResponse->json();
+        }
+
+        return [];
+    }
+
+
+    private function createUserFromResponse($request, $data): Model
+    {
+
+        $userService = App::make(UserServiceContract::class);
+        $userGroupService = App::make(UserGroupServiceContract::class);
+
+        $userSaveDto = new UserSaveDto($data['firstname'], $data['lastname'], true, [], $request->get('email'), $request->get('password'), true);
+
+        $userSettings = collect($data)->only(['city', 'postal_code', 'country', 'gender'])->toArray();
+
+        $mainGroup = Group::firstOrCreate([
+            'name' => 'Nauczyciele',
+            'registerable' => true
+        ]);
+
+        $groups = [$mainGroup];
+
+        $groups = array_merge($groups, $this->createOrFindGroups($mainGroup, $data['user_groups1']));
```

Comment:
Will data always have these three keys/values? (`user_groups1` etc)

Reply:
yes. it's hardcoded